lib/core/src/Cardano/Pool/DB.hs

```diff
@@ -29,7 +33,17 @@ import Data.Word
     ( Word64 )
 
 -- | A Database interface for storing pool production in DB.
-data DBLayer m = DBLayer
+--
+-- To use it, you will need the NamedFieldPuns extension and wrap operations
+-- with @atomically@:
+--
+-- Example:
+-- >>> :set -XNamedFieldPuns
+-- >>> DBLayer{atomically,putPoolProduction} = db
+-- >>> atomically $ putPoolProduction blockHeader pool
+--
+-- This gives you the power to also run /multiple/ operations atomically.
+data DBLayer m2 = forall m. MonadFail m => DBLayer
```

Comment:
I think we can get rid of `m2` altogether and simply force `IO` down there. 

Rationale: 

- It'll ease type-signatures on the call-sites
- We already do testing in IO for this anyway, might it be for the MVar implementation or the SQLite one
- The only sensible choice I could imagine for `m` apart from `IO` would be `STM`, which gives us very little value (none?) because we get the same guarantees running queries in `SQLPersistT` (what `m` really is in the end). 

All-in-all, parameterizing the `DBLayer` here doesn't buy us much.